librz/arch/p/analysis/analysis_x86_zydis.c

```diff
@@ -0,0 +1,3778 @@
+			strncpy(out, buf_, BUF_SZ);
+			component_count++;
+		}
+
+		if (base != "none") {
+			rz_strf(buf_, "%s%s,", out, base);
+			strncpy(out, buf_, BUF_SZ);
+			component_count++;
+		}
+
+		if (component_count > 1) {
+			if (component_count > 2) {
+				rz_strf(buf_, "%s+,", out);
+				strncpy(out, buf_, BUF_SZ);
+			}
+			if (disp < 0) {
+				rz_strf(buf_, "%s-", out);
+			} else {
+				rz_strf(buf_, "%s+", out);
+			}
+			strncpy(out, buf_, BUF_SZ);
+		} else {
+			// Remove the trailing ',' from esil statement.
+			if (*out) {
+				int out_len = strlen(out);
+				out[out_len > 0 ? out_len - 1 : 0] = 0;
+			}
+		}
+
+		// set = 2 is reserved for lea, where the operand is a memory address,
+		// but the corresponding memory is not loaded.
+		if (set == 1) {
+			rz_strf(buf_, "%s,%s=[%d]", out, setarg, op.size == 10 ? 8 : op.size);
+			strncpy(out, buf_, BUF_SZ);
+		} else if (set == 0) {
+			if (!*out) {
+				strcpy(out, "0");
+			}
+			rz_strf(buf_, "%s,[%d]", out, op.size == 10 ? 8 : op.size);
+			strncpy(out, buf_, BUF_SZ);
+		}
+		out[BUF_SZ - 1] = 0;
+		return out;
+	}
+	}
+	return NULL;
+}
+
+static int cond_x862r2(ZydisMnemonic mnemonic) {
+	// TODO : Some cases to be handled
+	switch (mnemonic) {
+	case ZYDIS_MNEMONIC_JZ: // Should be JE but JE is alias of JZ
+		return RZ_TYPE_COND_EQ;
+	case ZYDIS_MNEMONIC_JNZ:
+		return RZ_TYPE_COND_NE;
+	case ZYDIS_MNEMONIC_JB:
+	case ZYDIS_MNEMONIC_JL:
+		return RZ_TYPE_COND_LT;
+	case ZYDIS_MNEMONIC_JBE:
+	case ZYDIS_MNEMONIC_JLE:
+		return RZ_TYPE_COND_LE;
+	case ZYDIS_MNEMONIC_JNBE:
+		return RZ_TYPE_COND_GT;
+	case ZYDIS_MNEMONIC_JNLE:
+		return RZ_TYPE_COND_GE;
+	case ZYDIS_MNEMONIC_JNB:
+		return RZ_TYPE_COND_GE;
+	case ZYDIS_MNEMONIC_JS:
+	case ZYDIS_MNEMONIC_JNS:
+	case ZYDIS_MNEMONIC_JO:
+	case ZYDIS_MNEMONIC_JNO:
+	case ZYDIS_MNEMONIC_JNL:
+	case ZYDIS_MNEMONIC_JP:
+	case ZYDIS_MNEMONIC_JNP:
+	case ZYDIS_MNEMONIC_JCXZ:
+	case ZYDIS_MNEMONIC_JECXZ:
+		break;
+	}
+	return 0;
+}
+
+/* reg indices are based on Intel doc for 32-bit ModR/M byte */
+static const char *reg32_to_name(ut8 reg) {
+	const char *const names[] = { "eax", "ecx", "edx", "ebx", "esp", "ebp", "esi", "edi" };
+	return reg < RZ_ARRAY_SIZE(names) ? names[reg] : "unk";
+}
+
+static void anop_esil(RzAnalysis *a, RzAnalysisOp *op, ut64 addr, const ut8 *buf, int len, ZydisDecodedInstruction *zydecode, ZydisDecodedOperand *zydeop) {
+	int rs = a->bits / 8;
+	const char *pc = (a->bits == 16) ? "ip" : (a->bits == 32) ? "eip"
+								  : "rip";
+	const char *sp = (a->bits == 16) ? "sp" : (a->bits == 32) ? "esp"
+								  : "rsp";
+	const char *bp = (a->bits == 16) ? "bp" : (a->bits == 32) ? "ebp"
+								  : "rbp";
+	const char *si = (a->bits == 16) ? "si" : (a->bits == 32) ? "esi"
+								  : "rsi";
+	struct Getarg gop = {
+		.zydecode = zydecode,
+		.zydeop = zydeop,
+		.bits = a->bits
+	};
+	char *src;
+	char *src2;
+	char *dst;
+	char *dst2;
+	char *dst_r;
+	char *dst_w;
+	char *dstAdd;
+	char *arg0;
+	char *arg1;
+	char *arg2;
```

Comment:
Init with `NULL` and make them `const`, since you assign const char pointers to them later. If you can't because `getarg` must return a `char *` from the heap, then you need to duplicate the `const char *` returned by Zydis. So you can free these here later.
Or don't assign Zydis reg names to them anywhere and pass them instead directly to `esilprintf()`